src/extensions/extension-loader/extension/extension.injectable.ts

```diff
@@ -21,19 +22,39 @@ const extensionInjectable = getInjectable({
 
       instantiate: (childDi) => {
         const extensionRegistrators = childDi.injectMany(extensionRegistratorInjectionToken);
+        const disposers: IReactionDisposer[] = [];
 
         return {
           register: () => {
-            const injectables = extensionRegistrators.flatMap((getInjectablesOfExtension) =>
-              getInjectablesOfExtension(instance),
-            );
+            extensionRegistrators.forEach((getInjectablesOfExtension) => {
+              const injectables = getInjectablesOfExtension(instance);
 
-            runInAction(() => {
-              childDi.register(...injectables);
+              disposers.push(
+                // injectables is either an array or a computed array, in which case
+                // we need to update the registered injectables with a reaction every time they change 
+                reaction(
+                  () => Array.isArray(injectables) ? injectables : injectables.get(),
+                  (currentInjectables, previousInjectables) => {
+                    // On the second reaction remove the previously registered injectables to avoid duplicate injectables
+                    if (previousInjectables) {
+                      childDi.deregister(...previousInjectables);
+                    }
+
+                    childDi.register(...currentInjectables);
+                  }, {
+                    fireImmediately: true,
+                  },
+                ));
             });
           },
 
           deregister: () => {
+            disposers.forEach(dispose => {
+              dispose();
+            });
+
+            disposers.length = 0;
```

Comment:
nit: Please use our `disposer` helper function to clean this up

Reply:
Fixed